### ISSUE TITLE
BUG: `fillna` specify duplicated column that is MultiIndex causes RecursionError

### ISSUE THREAD

**Issue by yuji38kwmt** [pandas-dev/pandas#53498]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas
import numpy

data = [[numpy.nan, 2, 3], [4, numpy.nan, 6], [7, 8, numpy.nan]]
df = pandas.DataFrame(
    data, columns=pandas.MultiIndex.from_tuples([("x", "a"), ("x", "a"), ("y", "b")])
)
df.fillna({("x", "a"): 0})
# RecursionError occurred
```

### Issue Description

I have created a dataframe that contains `MultiIndex` columns. However, the columns in the dataframe are duplicated.

When I execute the `fillna` function and specify the duplicated column, a RecursionError occurs.

### Expected Behavior

I expected that RecursionError did not occur.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 965ceca9fd796940050d6fc817707bba1c4f9bff
python           : 3.11.2.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.15.0-72-generic
Version          : #79-Ubuntu SMP Wed Apr 19 08:22:18 UTC 2023
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : ja_JP.UTF-8
LOCALE           : ja_JP.UTF-8

pandas           : 2.0.2
numpy            : 1.24.2
pytz             : 2022.7.1
dateutil         : 2.8.2
setuptools       : 65.5.0
pip              : 23.1.2
Cython           : None
pytest           : 7.3.1
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.9.2
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.11.0
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : None
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 11.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : None
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
zstandard        : None
tzdata           : 2023.3
qtpy             : None
pyqt5            : None

</details>

**Comment by JostBrand** [comment]
I don't think this is a bug and more of a feature request for fillna to support [MultiIndex.](https://github.com/pandas-dev/pandas/blob/aefede93b1c2b0a2e311220bfdd8c1af97a92441/pandas/core/indexes/multi.py#L1617C19-L1617C19):
```
    def fillna(self, value=None, downcast=None):

        """
        fillna is not implemented for MultiIndex
        """
```

So the question remains why you get a recursion error and not a NotImplementedError. But as the user [documentation](https://pandas.pydata.org/docs/user_guide/duplicates.html) points out for duplicate labels:  

> [...] Other methods, like indexing, can give very surprising results.

I know this solution `df['x']['a'].fillna(0)` is not really satisfying.
